Kernel/Process.h

```diff
@@ -153,6 +160,21 @@ class Process final
         return callback(m_protected_values_do_not_access_directly);
     }
 
+    ErrorOr<FlatPtr> with_mutable_protected_data_for_new_privs(auto&& callback)
+    {
+        SpinlockLocker locker(m_protected_data_lock);
+        unprotect_data();
+        auto guard = ScopeGuard([&] { protect_data(); });
+        // NOTE: If no_new_privs was set via a prctl syscall, disallow any further code from doing
+        // any change!
+        if (m_protected_values_do_not_access_directly.no_new_privs_mode == NoNewPrivsMode::Enforced)
+            return EPERM;
+        if (m_protected_values_do_not_access_directly.no_new_privs_mode == NoNewPrivsMode::EnforcedQuietly)
+            return 0;
```

Comment:
What is the motivation behind silently allowing setuid and friends to fail?

>The second mode resembles the Linux behavior, and is useful for testing
on userspace code that might not be adapted to the mentioned conditions
being effective.

I don't think we need to care much about adapting to userspace code that is not familiar with the behavior. If code expects to run as root, or some other user, and it can't, I don't see a good reason to let it pretend that everything is fine when it's not. We control the whole stack here.

Reply:
The idea was to allow testing to see other kind of failures and detect them easily, it was not intended that we keep compatibility with Linux to import Linux userland applications to our ecosystem that use the no_new_privs mechanism.
Also, the idea was to allow using this mechanism on a lot more userland code right now so this mechanism will be more effective - for example, running `pls id` with the first mode will just spit an error, but with the second mode, will let it run. This is a small example, but the implication is huge - it means that the kernel can support with this mechanism being effective many of our userland code without introducing changes in the userspace side.